engine/src/main/java/org/terasology/rendering/nui/layouts/FlowLayout.java

```diff
@@ -54,14 +62,15 @@ public void onDraw(Canvas canvas) {
         int heightOffset = 0;
         for (UIWidget widget : contents) {
             Vector2i size = canvas.calculatePreferredSize(widget);
-            if (filledWidth != 0 && filledWidth + size.x  > canvas.size().x) {
+            if (filledWidth != 0 && filledWidth + size.x > canvas.size().x) {
                 heightOffset += filledHeight;
```

Comment:
I think the vertical spacing should be added here, instead:
```suggestion
                heightOffset += filledHeight;
                heightOffset += verticalSpacing;
```

This part of the code resets the filled width and height after increasing the height offset, e.g., after wrapping the line.

Reply:
I see I see, that seems like it, yes. Placed it in that position at the very start (because the other `filledHeight` was there, not because I had the wrapping in mind), but among my other changes till I reach the correct solution I changed it. I'll commit the suggestion 👍